src/app/main.cpp

```diff
@@ -346,6 +387,91 @@ void displayBadArgMessage(const QString &message)
 #endif
 }
 
+namespace
+{
+#ifndef Q_OS_WIN
+/*
+ * Borrowed from https://stackoverflow.com/a/758054
+ *
+ * getexename - Get the filename of the currently running executable
+ *
+ * The getexename() function copies an absolute filename of the currently
+ * running executable to the array pointed to by buf, which is of length size.
+ *
+ * If the filename would require a buffer longer than size elements, NULL is
+ * returned, and errno is set to ERANGE; an application should check for this
+ * error, and allocate a larger buffer if necessary.
+ *
+ * Return value:
+ * NULL on failure, with errno set accordingly, and buf on success. The
+ * contents of the array pointed to by buf are undefined on error.
+ *
+ * Notes:
+ * This function is tested on Linux only. It relies on information supplied by
+ * the /proc file system.
+ * The returned filename points to the final executable loaded by the execve()
+ * system call. In the case of scripts, the filename points to the script
+ * handler, not to the script.
+ * The filename returned points to the actual exectuable and not a symlink.
+ *
+ */
+char *getExeName(char *buf, std::size_t size)
+{
+#ifdef NO_PROC_SELF_DIR
+    char linkname[64]; // /proc/<pid>/exe
+
+    // Get our PID and build the name of the link in /proc
+    pid_t pid = ::getpid();
+
+    if (snprintf(linkname, sizeof(linkname), "/proc/%i/exe", pid) < 0) {
+        qDebug() << "Size of integer decimal representation does not fit into 54 bytes long fixed buffer";
+        errno = E2BIG;
+        return nullptr;
+    }
+
+    // Now read the symbolic link
+    ssize_t ret = readlink(linkname, buf, size);
+#else
+    ssize_t ret = ::readlink("/proc/self/exe", buf, size);
+#endif
+
+    // In case of an error, leave the handling up to the caller
+    if (ret == -1)
+        return nullptr;
+
+    // Report insufficient buffer size
+    if (static_cast<std::size_t>(ret) >= size) {
+        errno = ERANGE;
+        return nullptr;
+    }
+
+    // Ensure proper null-termination
+    buf[ret] = 0;
+    return buf;
+}
+#endif
+}
+
+QString executablePath()
+{
+#ifdef Q_OS_WIN
+    WCHAR apath[MAX_PATH];
+    ::GetModuleFileNameW(NULL, apath, MAX_PATH);
+    return QString::fromWCharArray(apath);
+#else
+    std::unique_ptr<char[]> buf = nullptr;
+    for (std::size_t size = 32; buf.reset(new char[size]), !getExeName(buf.get(), size); size *= 2) {
```

Comment:
Why not put this logic in getExeName? I mean buffer asjusting.
And why not create it of MAX_PATH size initially?

Reply:
Because MAX_PATH is purely Windows limitation, and on UNIX most executable reside /usr/bin or /bin, thus making the first guess (32) work almost always.